haplotypecaller-gvcf-gatk4.wdl

```diff
@@ -87,12 +102,51 @@ workflow HaplotypeCallerGvcf_GATK4 {
 
 # TASK DEFINITIONS
 
+task CramToBamTask {
+  # Command parameters
+  File ref_fasta
+  File ref_fasta_index
+  File ref_dict
+  File input_cram
+  String sample_name
+
+  # Runtime parameters
+  String docker
+  Int? machine_mem_gb
+  Int? disk_space_gb
+  Boolean use_ssd = false
+  Int? preemptible_attempts
+
+  Float output_bam_size = size(input_cram, "GB") / 0.60
+  Float ref_size = size(ref_fasta, "GB") + size(ref_fasta_index, "GB") + size(ref_dict, "GB")
+  Int disk_size = ceil(size(input_cram, "GB") + output_bam_size + ref_size) + 20
+
+  command {
+    set -e
+    set -o pipefail
+
+    samtools view -h -T ${ref_fasta} ${input_cram} |
```

Comment:
I am wondering why a cram to bam task has been added. I was able to successfully run this wdl (pre CramtoBam) with a cram file as input. HaplotypeCaller seemed to handle this fine. 

However there were some minor numerical differences in the output (I think they were minor, I am not sure). I asked about this here: [gatkforums](https://gatkforums.broadinstitute.org/gatk/discussion/11988/cram-vs-bam-difference-in-output-of-haplotypecaller#latest) However I have not had a chance to get to the bottom of the issue.

Beside that point, there is also an issue with samtools vs gatk calculation of the NM and MD values. [samtools](https://github.com/samtools/hts-specs/pull/308) . If the cram does not contain NM and MD (default cram behaviour), then samtools will use its own method to calculate those values. The method is soon to be standardised in [hts-specs](https://github.com/samtools/hts-specs/pull/308), but I'm not sure how gatk plans to resolve the difference. 

Lastly the pipe is unnecessary, you can directly add -b -o to the first samtools line.

Reply:
There are a few bugs with the code used to process cram files that our developers are in the process of fixing along with a number of critical performance issues remaining. Thus the following task was added to convert cram files to bam files before processing with HaplotypeCaller.
Thanks for the feedback, I've directed the developers to this comment and the forum post you mentioned.